data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/CertificateProvider.java

```diff
@@ -0,0 +1,8 @@
+package com.amazon.dataprepper.plugins.certificate;
+
+import com.amazon.dataprepper.plugins.certificate.model.Certificate;
+import com.amazon.dataprepper.plugins.source.oteltrace.OTelTraceSourceConfig;
+
+public interface CertificateProvider {
+    Certificate getCertificate(final OTelTraceSourceConfig oTelTraceSourceConfig);
```

Comment:
Just curious - can this signature be changed to accept `(String, String)` (or whatever else may be needed) instead of the full Config object? Ideally we'd pass exactly what the provider needs to create a Certificate instead of relying on it knowing how to navigate the Config object.

Update: After looking at more of the PR, maybe this can accept  no arguments and rely on the instance variables of the provider (e.g. pass file paths to FileCertificateProvider's constructor).

Reply:
yeah, we can make it no argument constructor. I will make the change.